src/overlays/actors/ovl_Obj_Tokeidai/z_obj_tokeidai.c

```diff
@@ -50,87 +42,803 @@ const ActorInit Obj_Tokeidai_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokeidai/func_80AB4394.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokeidai/func_80AB4664.s")
+// static InitChainEntry sInitChain
+static InitChainEntry D_80AB49E0[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 3300, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1100, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokeidai/func_80AB4894.s")
+extern Gfx D_06009A08[]; // the tower's four walls, used in Termina Field
+extern Gfx D_0600B0C0[]; // spotlight emitted from the counterweight at night
+extern Gfx D_0600B208[]; // the tower's counterweight
+extern Gfx D_0600BA78[]; // rotating gear on the side of the tower
+extern Gfx D_0600BEE8[]; // clock center and "hand"
+extern Gfx D_0600C368[]; // sun and moon disk
+extern Gfx D_0600CF28[]; // outer ring of the clock face
+extern Gfx D_0600D388[]; // unused tower wall
+extern Gfx D_0600D8E0[]; // empty dlist
+extern Gfx D_0600D8E8[]; // spiral staircase to tower interior
+extern Gfx D_0600E818[]; // clock tower clock face
+extern Gfx D_0600F518[]; // wall clock clock face
+
+/**
+ * Returns the angle necessary to show the correct side of
+ * the sun and moon disk based on the time of day. The actual
+ * angle can differ from the target angle if the disk is in
+ * the middle of rotating.
+ */
+s32 ObjTokeidai_GetTargetSunMoonDiskRotation() {
+    if (gSaveContext.isNight) {
+        return 0x8000;
+    }
+    return 0;
+}
+
+void ObjTokeidai_SetupClockOrGear(ObjTokeidai* this) {
+    s32 currentMinute;
+
+    currentMinute = OBJ_TOKEIDAI_GET_CURRENT_MINUTE(this);
+    this->clockMinute = currentMinute;
+    this->outerRingOrGearRotation = OBJ_TOKEIDAI_GET_OUTER_RING_OR_GEAR_ROTATION(currentMinute);
+    this->outerRingOrGearRotationalVelocity = 0x3C;
+    this->outerRingOrGearRotationTimer = 0;
+}
+
+void ObjTokeidai_Clock_Init(ObjTokeidai* this) {
+    s32 currentHour;
+
+    ObjTokeidai_SetupClockOrGear(this);
+    currentHour = OBJ_TOKEIDAI_GET_CURRENT_HOUR(this);
+    this->clockHour = currentHour;
+    this->clockFaceRotation = OBJ_TOKEIDAI_GET_CLOCK_FACE_ROTATION(currentHour);
+    this->clockFaceRotationalVelocity = 0;
+    this->clockFaceRotationTimer = 0;
+    this->sunMoonDiskRotationalVelocity = 0;
+    this->sunMoonDiskRotation = ObjTokeidai_GetTargetSunMoonDiskRotation();
+}
+
+void ObjTokeidai_TowerGear_Init(ObjTokeidai* this, GlobalContext* globalCtx) {
+    this->actor.draw = ObjTokeidai_TowerGear_Draw;
+    this->opaDList = D_0600BA78;
+    ObjTokeidai_SetupClockOrGear(this);
+
+    if ((globalCtx->sceneNum == SCENE_CLOCKTOWER && gSaveContext.sceneSetupIndex == 2 &&
+         globalCtx->csCtx.unk_12 == 0) ||
+        (globalCtx->sceneNum == SCENE_00KEIKOKU && gSaveContext.sceneSetupIndex == 2 && globalCtx->csCtx.unk_12 == 0)) {
```

Comment:
```suggestion
    if (((globalCtx->sceneNum == SCENE_CLOCKTOWER) && (gSaveContext.sceneSetupIndex == 2) &&
         (globalCtx->csCtx.unk_12 == 0)) ||
        ((globalCtx->sceneNum == SCENE_00KEIKOKU) && (gSaveContext.sceneSetupIndex == 2) && (globalCtx->csCtx.unk_12 == 0))) {
```
I feel this is easier to read once you get to the big if-blocks, but it's personal preference. So you can disregard

Reply:
I realized after this comment that they probably didn't write the same gigantic if-block four times; it was probably just a macro. So if you still feel this way about the resulting macro, just let me know.

Reply:
Maybe it was, maybe it wasn't. No clue. There are definitely if-block much larger than this that are much more messier. Plus I find macros tend to have even more brackets than even what's in code. Again, it's personal preference for me with it but something like this is up to the decompers discretion :)

Reply:
When I removed the macro and put these in code, I added these parentheses.